File: server - Fast Api/models/BookModel.py
from pydantic import BeforeValidator, EmailStr, BaseModel, Field
from typing import Annotated, Optional
# from bson import ObjectId

PyObjectId = Annotated[str, BeforeValidator(str)]


class Book(BaseModel):
    name: str = Field(...) # Book title
    isbn: str = Field(...)
    author: str = Field(...)
    publisher: str = Field(...)
    year: str = Field(...)
    genre: str = Field(...)
    quantity: str = Field(...)
    status: str = Field(...)

File: backend/model/bookModel.ts
import mongoose from "mongoose";
export const bookSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    required: [true, "Please provide user id"],
  },
  name: {
    type: String,
    required: [true, "Please enter valid email id"],
  },
  isbn: {
    type: String,
    required: [true, "Please enter valid email id"],
    unique: true,
  },
  author: {
    type: String,
    required: [true, "Please enter valid email id"],
  },
  publisher: {
    type: String,
    required: [true, "Please enter valid email id"],
  },
  year: {
    type: Number,
    required: [true, "Please enter valid email id"],
  },
  genre: {
    type: String,
    required: [true, "Please enter valid email id"],
  },
  quantity: {
    type: Number,
    required: [true, "Please enter valid email id"],
  },
  status: {
    type: String,
    enum: ["available", "not-available"],
    default: "available",
    required: true,
  },
});
export const Book = mongoose.model("Books", bookSchema);